ABDataCollectionCore.js

```diff
@@ -1551,13 +1628,30 @@ module.exports = class ABDataCollectionCore extends ABMLClass {
                const currentCursor = this.getCursor();
 
                // If the cursor is not the new, then it should not reload.
-               if (followCursor?.[followDC.datasource.PK()] == currentCursor?.[this.datasource.PK()])
+               if (
+                  followCursor?.[followDC.datasource.PK()] ==
+                  currentCursor?.[this.datasource.PK()]
+               )
                   return;
 
                this.clearAll();
                this.loadData();
             },
          });
+
+         this.eventAdd({
+            emitter: followDC,
+            eventName: "cursorStale",
+            listener: () => {
+               // cursorStale : the current cursor hasn't CHANGED, but the data
+               // of that value has changed.
+               // This is triggered by one of our socket updates that detects
+               // changes to the cursor data.
+
+               this.clearAll();
+               this.loadData();
```

Comment:
If we will change the way to update data of child DCs, then we should to remove this webix event ?
https://github.com/digi-serve/ab_platform_web/blob/master/AppBuilder/platform/ABDataCollection.js#L119

This event is triggered when webix.DataCollection get change cursor.

To prevent redundancy load data.

Reply:
That's a good catch, I hadn't looked there.

However, in that routine they are checking the change in values before they perform a `.reloadData()`:
```
if (dc.__prevLinkDcCursor != value) {
   cursorUpdated = true;
}
```

In our new `cursorStale` updates, the `.id` isn't changing and as a result, this `datafeed` event wouldn't update the data unless we have a new row with a different `.id`.

So in the end, I don't think we will have a double load as a result.